src/containers/ProfilePage.jsx

```diff
@@ -89,9 +96,24 @@ class ProfilePage extends React.Component {
     });
   }
 
+  checkPermissions() {
+    const { invalidPermission } = this.state;
+    if (Config.b2b.enable && invalidPermission) {
+      return (
+        <div className="message-permission">
+          <h2>{intl.get('permission-message')}</h2>
+        </div>
+      );
+    }
+    return (
+      <div className="loader" />
+    );
+  }
+
   render() {
     const { profileData } = this.state;
-    const email = profileData && profileData._emails[0]._element ? profileData._emails[0]._element[0].email : '';
+    const email = profileData && profileData._emails && profileData._emails[0]._element ? profileData._emails[0]._element[0].email : '';
```

Comment:
The profile page is still currently broken with this implementation. We need to check if resources are available in the API before displaying them.